clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -256,14 +253,14 @@ private void process(final UnsubscribeEvent event) {
         }
     }
 
-    private void process(final ResetPositionsEvent event) {
-        CompletableFuture<Void> future = requestManagers.offsetsRequestManager.resetPositionsIfNeeded();
-        future.whenComplete(complete(event.future()));
-    }
-
-    private void process(final ValidatePositionsEvent event) {
-        CompletableFuture<Void> future = requestManagers.offsetsRequestManager.validatePositionsIfNeeded();
-        future.whenComplete(complete(event.future()));
+    /**
+     * Fetch committed offsets and use them to update positions in the subscription state. If no
+     * committed offsets available, fetch offsets from the leader.
+     */
+    private void process(final UpdateFetchPositionsEvent updateFetchPositionsEvent) {
+        CompletableFuture<Boolean> future =
+            requestManagers.offsetsRequestManager.updateFetchPositions(updateFetchPositionsEvent.deadlineMs());
+        future.whenComplete(complete(updateFetchPositionsEvent.future()));
```

Comment:
Super nit: any reason not to make the variable named `event` like the other methods? That'll help reduce visual clutter and make the lines a bit shorter:

```suggestion
    private void process(final UpdateFetchPositionsEvent event) {
        CompletableFuture<Boolean> future =
            requestManagers.offsetsRequestManager.updateFetchPositions(event.deadlineMs());
        future.whenComplete(complete(event.future()));
```